exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/union/UnionAllRecordBatch.java

```diff
@@ -162,6 +162,25 @@ private IterOutcome doWork() throws ClassTransformationException, IOException, S
     allocationVectors = Lists.newArrayList();
     transfers.clear();
 
+    // If both sides of Union-All are empty
+    if(unionAllInput.isBothSideEmpty()) {
+      for(int i = 0; i < outputFields.size(); ++i) {
```

Comment:
If both sides are empty, wouldn't outputFields.size() be 0 ?  Or is outputFields populated from a previous batch, which implies both sides may not have been empty.  Can you clarify ?